appinventor/blocklyeditor/src/versioning.js

```diff
@@ -747,22 +747,20 @@ Blockly.Versioning.v17_translateComponentSetGetProperty = function(blockElem) {
  Upgrade screen names to use dropdown block.
  ******************************************************************************/
 
-Blockly.Versioning.makeScreenNamesBeDropdowns = function () {
-  return function(blocksRep, workspace) {
-    var dom = Blockly.Versioning.ensureDom(blocksRep);
-    for (var i = 0, block; block = dom.children[i]; i++) {
-      if (block.tagName == 'block' &&
-          (block.getAttribute('type') == 'controls_openAnotherScreen' ||
-          block.getAttribute('type') == 'controls_openAnotherScreenWithStartValue')) {
-        var value = Blockly.Versioning.firstChildWithTagName(block, 'value');
-        var name = value.getAttribute('name');
-        if (name == 'SCREENNAME' || name == 'SCREEN') {
-          Blockly.Versioning.tryReplaceBlockWithScreen(value);
-        }
+Blockly.Versioning.makeScreenNamesBeDropdowns = function (blocksRep, workspace) {
+  var dom = Blockly.Versioning.ensureDom(blocksRep);
+  for (var i = 0, block; block = dom.children[i]; i++) {
```

Comment:
I believe this will only get you the top level blocks. You'll want to do `blocksRep.getElementsByTagName('block')` to get all of the blocks. You may want to test this upgrader with a slightly less trivial nesting of blocks to make sure.

Reply:
Gosh that's a scary bug. Thank's for the catch! Fixed & tested :D